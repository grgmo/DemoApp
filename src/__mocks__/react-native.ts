import RN from 'react-native';

RN.NativeModules.AppInfo = {
  getConstants: () => ({
    appVersion: '1.0.0',
  }),
};
module.exports = RN;
