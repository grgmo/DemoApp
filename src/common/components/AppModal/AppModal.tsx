import React, {FC, ReactNode} from 'react';
import {
  Button,
  Modal,
  ModalProps,
  SafeAreaView,
  StyleSheet,
} from 'react-native';

interface AppModalType extends ModalProps {
  children?: ReactNode;
}

const AppModal: FC<AppModalType> = ({children, onRequestClose, ...props}) => {
  return (
    <Modal
      testID="modal"
      animationType="slide"
      onRequestClose={onRequestClose}
      {...props}>
      <SafeAreaView style={styles.container}>
        <>
          <Button testID="close_modal" title="Close" onPress={onRequestClose} />
          {children}
        </>
      </SafeAreaView>
    </Modal>
  );
};

export default AppModal;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
