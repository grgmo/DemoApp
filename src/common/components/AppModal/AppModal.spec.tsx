import React from 'react';
import {View} from 'react-native';
import {fireEvent, render} from '@testing-library/react-native';

import AppModal from './AppModal';

describe('AppModal', () => {
  it('should render modal correctly', () => {
    const modal = render(
      <AppModal>
        <View />
      </AppModal>,
    );
    expect(modal.toJSON()).toMatchSnapshot();
  });

  it('should call onRequestClose on close button press', () => {
    const mockOnRequestClose = jest.fn();
    const {getByRole} = render(
      <AppModal onRequestClose={mockOnRequestClose} />,
    );
    const button = getByRole('button');

    fireEvent.press(button);

    expect(mockOnRequestClose).toHaveBeenCalledTimes(1);
  });

  it('should call onRequestClose on onRequestClose', () => {
    const mockOnRequestClose = jest.fn();
    const {getByTestId} = render(
      <AppModal onRequestClose={mockOnRequestClose} />,
    );
    const {props} = getByTestId('modal');
    props.onRequestClose();

    expect(mockOnRequestClose).toHaveBeenCalledTimes(1);
  });
});
