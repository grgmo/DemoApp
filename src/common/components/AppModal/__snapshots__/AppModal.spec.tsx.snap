// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal should render modal correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  testID="modal"
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Close
        </Text>
      </View>
    </View>
    <View />
  </RCTSafeAreaView>
</Modal>
`;
